### PR TITLE
Set TTLs During Closure Execution

### PR DESCRIPTION
**Pull request by baileylo**
Allows developers to set different TTLs based on the result of the cache look up.

### CODE REVIEW

File: src/OffloadManager.php [hunk 1]
```diff
@@ -208,6 +208,8 @@ protected function run($key, $task, $options)
 		if ($ttl_fresh || $ttl_stale) {
 			$result->then(function ($data) use ($key, $run, $ttl_fresh, $ttl_stale) {
 				if (!$run->isBad()) {
+					$ttl_fresh = $run->getTtlFresh() ?: $ttl_fresh;
```
Comment (baileylo) [inline src/OffloadManager.php]: Maybe this should be a `null` check. Maybe somebody wants to set something to 0 on purpose.

File: src/OffloadRun.php [hunk 2]
```diff
@@ -8,6 +8,10 @@ class OffloadRun
 	protected $bad = false;
 	/** @var array Additional cache options. */
 	protected $cache_options = [];
+	/** @var float|null NUll values are ignored, number of seconds to cache value  */
```
Comment (ralouphie) [inline src/OffloadRun.php]: `NUll` => `Null`

File: src/OffloadRun.php [hunk 3]
```diff
@@ -8,6 +8,10 @@ class OffloadRun
...
+	protected $ttl_fresh;
+	/** @var float|null NUll values are ignored, number of seconds set stale cache */
```
Comment (ralouphie) [inline src/OffloadRun.php]: `NUll` => `Null`

File: src/OffloadRun.php [hunk 4]
```diff
@@ -42,4 +46,36 @@ public function getCacheOptions()
...
+	/**
+	 * @return float|null Null to use default TTLs
+	 */
+	public function getTtlFresh()
+	{
+		return $this->ttl_fresh;
+	}
+
+	/**
+	 * @param float|null $ttl_fresh
```
Comment (ralouphie) [inline src/OffloadRun.php]: Missing description.
`float|null $ttl_fresh $ttl_fresh The TTL to use for the fresh time period (null for default).`

File: src/OffloadRun.php [hunk 5]
```diff
@@ -42,4 +46,36 @@ public function getCacheOptions()
...
+	 */
+	public function setTtlFresh($ttl_fresh)
+	{
+		$this->ttl_fresh = $ttl_fresh;
+	}
+
+	/**
+	 * @return float|null
```
Comment (ralouphie) [inline src/OffloadRun.php]: Missing description.

File: src/OffloadRun.php [hunk 6]
```diff
@@ -42,4 +46,36 @@ public function getCacheOptions()
...
+	 */
+	public function getTtlStale()
+	{
+		return $this->ttl_stale;
+	}
+
+	/**
+	 * @param float|null $ttl_stale
```
Comment (ralouphie) [inline src/OffloadRun.php]: Missing description.

File: src/OffloadManager.php (hunk 1 above)
Comment (ralouphie) [inline src/OffloadManager.php]: Yeah, a `null` check would be better.

**Comment by ralouphie** [comment]
@baileylo Can you add tests for this and bring code coverage back to 100%

**Pull request merged**